### DISCUSSION TITLE
fetch data when a local component state change?

### DISCUSSION THREAD

**Discussion by Adesh66** [TanStack/query#3276] [Q&A]
Hey 
I wanted to call my get query method only when a local state of my component change. let suppose a dropdown value change and in that I want to trigger my get query call.

Thanks

**Comment by TkDodo** [comment]
Make the local state part of the query key.

**Reply by Adesh66** [reply]
will it work with rtk query as well in the same way ?

**Reply by TkDodo** [reply]
I have not used rtk-query so I cannot comment on that. You'd have to ask the redux team :)
